### PR TITLE
[CI][Bugfix][AMD][ Ensure weights created when using emulating OCP MXFP4

### PR DESCRIPTION
**Pull request by rasmith**
## Purpose
The `test_ocp_mx_moe.py:test_mxfp4`_loading_and_execution_moe fails due to two reasons:
1) Setting `cuda_graph_capture_sizes` in the runner causes a `ValueError` to be thrown because setting them in the runner and in the compilation config are mutually exclusive. This is because `VllmRunner` already sets the `cudagraph_capture_sizes` in its compilation config.
    * Setting this in the compilation config in the test fixes the problem.
 2) Weights are not properly created when running in emulation mode on MI300. This results in the test crashing when `process_weights_after_loading` which will try to process a `weight_scale` that hasn't been created during `create_weights`.
     * Ensuring weights are created in the quark ocp mx scheme with emulation mode fixes the problem.

## Test Plan
`pytest -sv test_ocp_mx_moe.py`
## Test Result
`5 passed, 116 skipped, 3 warnings in 219.96s (0:03:39)`

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [X ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ X] The test plan, such as providing test command.
- [ X] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py [hunk 1]
```diff
@@ -316,51 +314,36 @@ def create_weights(
         weight_loader: Callable,
         **kwargs,
     ):
-        if self.dynamic_mxfp4_quant:
```
Comment (tjtanaa) [inline vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:317]: @BowenBao Could you take a look at this ?

Comment (BowenBao) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:317]: @dllehr-amd for review, I believe this is for the dynamic mxfp4 quant during loading?

File: vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py
Comment (BowenBao) [inline vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:280]: perhaps the right fix is to port `if self.dynamic_mxfp4_quant:` logic here as well, such that weight and scales are properly populated for emulation too.

@dllehr-amd when is `dynamic_mxfp4_quant` turned on?

Comment (dllehr-amd) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:280]: @BowenBao   it's used when we detect that we're on a deepseek_v3 model. in `maybe_update_config` in quark.py

File: vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py (hunk 1 above)
Comment (dllehr-amd) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:317]: @tjtanaa   This is used for deepseek_v3 models, we're quantizing the attn proj weights after loading them in BF16.  This shouldn't have an affect on this test though.

**Review by BowenBao — CHANGES_REQUESTED** [review]
Putting a hold on this PR as it might break the `dynamic_mxfp4_quant` feature @dllehr-amd added earlier.

I believe the right fix is to [do this](https://github.com/vllm-project/vllm/pull/36993/changes#r2955947754) and revert [the current PR's change in this section](https://github.com/vllm-project/vllm/pull/36993/changes#r2950259986).

iiuc w/o this pr it is only failing `fxmarty/deepseek_r1_3_layers_mxfp4` subcase in test_mxfp4_loading_and_execution_moe on mi300, since it's emulate code flow and triggers dynamic_mxfp4_quant

File: vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py
Comment (BowenBao) [inline vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:280]: @rasmith something like this
it should work if `dynamic_mxfp4_quant` does not shuffle weight or scales
otherwise the quickest fix would be to disable `dynamic_mxfp4_quant` if emulation.

```python
        if self.emulate:
            if self.dynamic_mxfp4_quant:
                w_q, w_s = dynamic_mxfp4_quant(layer.weight)
                layer.weight_scale = torch.nn.Parameter(
                    w_s.T.contiguous(), requires_grad=False
                )
                layer.weight = torch.nn.Parameter(w_q, requires_grad=False)
            else:
                layer.weight_scale = torch.nn.Parameter(
                    layer.weight_scale.data, requires_grad=False
                )
```

Comment (rasmith) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:280]: @BowenBao  @dllehr-amd  I did this and this also works.  Please take another look?

Comment (rasmith) [reply vllm/model_executor/layers/quantization/quark/schemes/quark_ocp_mx.py:280]: > @rasmith something like this it should work if `dynamic_mxfp4_quant` does not shuffle weight or scales otherwise the quickest fix would be to disable `dynamic_mxfp4_quant` if emulation.
> 
> ```python
>         if self.emulate:
>             if self.dynamic_mxfp4_quant:
>                 w_q, w_s = dynamic_mxfp4_quant(layer.weight)
>                 layer.weight_scale = torch.nn.Parameter(
>                     w_s.T.contiguous(), requires_grad=False
>                 )
>                 layer.weight = torch.nn.Parameter(w_q, requires_grad=False)
>             else:
>                 layer.weight_scale = torch.nn.Parameter(
>                     layer.weight_scale.data, requires_grad=False
>                 )
> ```

Yes, I basically did this.

**Review by BowenBao — APPROVED** [review]
LGTM, thank you @rasmith !

cc @tjtanaa

**Review by tjtanaa — APPROVED** [review]
Thanks @BowenBao for helping out to review Quark related PRs.
